Filtering special characters during automatic linking of ingredients …

jenscalaerts opened the pull request:
…to instructions

Used a unicode group to have a set of all unicode punctuation marks


## What type of PR is this?

- bug

## What this PR does / why we need it:

_(REQUIRED)_
The pull request aims to improve 

## Which issue(s) this PR fixes:
Fixes #1803 

## Testing
Create recipe with ingredients:
* 2 eggs
* 3 potatoes
* 5 carrots
* 2 steaks
Create an instruction with containing words matching ingredients, who have punctuation marks or formatting characters attached to them. For example, 
```
add the **eggs**  to the 
potatoes, then mix
```
Press "Link Ingredients" on this instruction then press the auto button.
eggs and potatoes should be selected. 
## Release Notes

_(REQUIRED)_
```
Fixed bug where steps were not automatically linked to ingredients when special characters are attached to the word.
```

jenscalaerts committed:
Filtering special characters during automatic linking of ingredients to instructions

Used a unicode group to have a set of all unicode punctuation marks

jenscalaerts committed:
allowing for linking of ingredients to instruction at the beginning of a newline in the instruction

hay-kot reviewed (changes requested):
Could you pull this function into a composable and provide a unit test for it? Other than that it looks great!

jenscalaerts committed:
Merge branch 'hay-kot:mealie-next' into fix/1803-instruction-recipe-linking-special-characters

jenscalaerts committed:
Extracted ingredient matching into a composable and added tests. Ignoring 2 letter words to avoid false matches.

While testing the code 2 letter matches were a large source of false positives.

jenscalaerts commented:
I have added filtering for 2 letter words in the instruction. They seemed to generate a large amount of false positives (in Dutch). This seems like a good idea because most languages have few 2 letter ingredients. It also replaces a large part of the blacklist, which has no internationalization.

hay-kot commented:
Excellent, thanks so much for following up with tests!
